Dockerfile

```diff
@@ -1,28 +1,19 @@
-# Multistage - Builder
-FROM maven:3.6.3-jdk-11-slim as s3proxy-builder
-LABEL maintainer="Andrew Gaul <andrew@gaul.org>"
-
-WORKDIR /opt/s3proxy
-COPY . /opt/s3proxy/
-
-RUN mvn package -DskipTests
-
-# Multistage - Image
 FROM openjdk:11-jre-slim
 LABEL maintainer="Andrew Gaul <andrew@gaul.org>"
 
+RUN mkdir /data ;\
+    useradd -rm -d /opt/s3proxy -s /bin/sh -u 1001 -U javauser
 WORKDIR /opt/s3proxy
 
 COPY \
-    --from=s3proxy-builder \
-    /opt/s3proxy/target/s3proxy \
-    /opt/s3proxy/src/main/resources/run-docker-container.sh \
+    target/s3proxy \
+    src/main/resources/run-docker-container.sh \
     /opt/s3proxy/
 
 ENV \
     LOG_LEVEL="info" \
     S3PROXY_AUTHORIZATION="aws-v2-or-v4" \
-    S3PROXY_ENDPOINT="http://0.0.0.0:80" \
+    S3PROXY_ENDPOINT="http://0.0.0.0:8000" \
```

Comment:
Yes I did mark the change breaking fixing the security risk needs to be secure by default but a user can change it back easily. So this would need to be a 3. Release

Reply:
What is the security risk?  I can't imagine breaking every single S3Proxy user given the millions on Docker pulls.

Reply:
Because docker isn't a vm the uid is mapped to the host and this means the app is running as root on the host. The risk is container escape which is a thing particularly with an app that has fs interactions by nature. An alternative would be to publish to flavors of the container leaving the current one as is but adding the breaking change with -rootless in the tag.